index.js

```diff
@@ -19,13 +19,73 @@ module.exports = robot => {
       head: pr.head.sha
     }))
 
-    const dcoParams = await getDCOStatus(compare.data.commits, requireMembers(requireForMembers, context))
+    const commits = compare.data.commits
+    const dcoFailed = await getDCOStatus(commits, requireMembers(requireForMembers, context), context.issue())
 
-    const params = Object.assign({
-      sha: pr.head.sha,
-      context: 'DCO'
-    }, dcoParams)
+    if (!dcoFailed.length) {
+      return context.github.checks.create(context.repo({
+        name: 'DCO',
+        head_branch: pr.head.ref,
+        head_sha: pr.head.sha,
+        status: 'completed',
+        conclusion: 'success',
+        completed_at: new Date(),
+        output: {
+          title: 'DCO',
+          summary: 'All commits are signed off!'
+        }
+      }))
+    } else {
+      let summary = []
+      dcoFailed.forEach(function(commit) {
+        summary.push(`Commit sha: [${commit.sha.substr(0, 7)}](${commit.url}), Author: ${commit.author}, Committer: ${commit.committer}; ${commit.message}`)
+      })
+      summary = summary.join('\n')
+      if (dcoFailed.length === 1) summary = handleOneCommit(pr, dcoFailed) + `\n\n${summary}`
+      else summary = handleMultipleCommits(pr, commits.length, dcoFailed) + `\n\n${summary}`
 
-    return context.github.repos.createStatus(context.repo(params))
+      return context.github.checks.create(context.repo({
+        name: 'DCO',
+        head_branch: pr.head.ref,
+        head_sha: pr.head.sha,
+        status: 'completed',
+        conclusion: 'action_required',
+        completed_at: new Date(),
+        output: {
+          title: 'DCO',
+          summary
+        },
+        actions: [{
+          label: 'Set DCO to pass',
+          description: 'would set status to passing',
```

Comment:
I'd love to get eyes on this from a legal perspective to make sure that a this is worded properly. Something like "Accept DCO" and "I certify that my contributions fall under the Developer Certificate of Origin: https://developercertificate.org/"

@caniszczyk @mlinksva: could you recommend someone to help here?

Also, I wonder if there should be a separate "Dismiss" vs "Accept" buttons

Reply:
So it's important to note this option is only accessible for users with write access to the repo. 

We get quite a few requests for new users to not have to use git so painfully, so if the maintainers of a repo are happy with the user's comments that they "accept" the DCO, might as well give them a free status pass button.

Reply:
Maybe I'm missing something, but this looks to me like a way for a maintainer to ignore the test, not for a contributor to agree their contributions are covered by the DCO.

Reply:
Doh, sorry, I totally wasn't thinking about who would be the one clicking the button.